### PR TITLE
Improve performance of ResourceManager::getResourceBySha1()

### PR DESCRIPTION
**Pull request by bwaidelich**
This adds an index to the `PersistentResource` table drastically
improving the performance of `ResourceManager::getResourceBySha1()`
calls (= public API).

### CODE REVIEW

File: TYPO3.Flow/Classes/TYPO3/Flow/Resource/Resource.php [hunk 1]
```diff
@@ -63,7 +64,7 @@ class Resource implements ResourceMetaDataInterface, CacheAwareInterface
      * The IANA media type of this resource
      *
      * @var string
-     * @Flow\Validate(type="StringLength", options={ "maximum"=100 })
+     * @Flow\Validate(type="StringLength", options={ "maximum"=100 })Version20180622074421.php
```
Comment (bwaidelich) [inline TYPO3.Flow/Classes/TYPO3/Flow/Resource/Resource.php]: oops, hang on

**Comment by bwaidelich** [comment]
I had to fix some typos.. (= Friday). Feel free to squash if you happen to merge this one

**Comment by kitsunet** [comment]
Mmm, I know I suggested 3.3, but I guess we need ot think about how we deal with upmerges and the migration. Because upmerging this migration will horribly fail ofc. So technically we would need to chdeck for table existence before adding the index and then add another migration after upmerging for the new table name. Or am I thinking too complicated here?

**Comment by kitsunet** [comment]
So, eh, or we just save ourselves the trouble and, uhm, go back again to 4.3? Sorry, just an idea, because I think this is very sensible to do and should go in ASAP.

**Comment by kdambekalns** [comment]
I'll take care of the migration voodoo today.

**Comment by bwaidelich** [comment]
> or we just save ourselves the trouble and, uhm, go back again to 4.3?

That was my thinking when I pushed this against 4.3 :)
But it should not be too much work to adjust this when upmerging (see #1300 for reference).

@kdambekalns Thanks for taking care!

**Comment by kitsunet** [comment]
Again, this is IMHO broken like this if you up merge. We need another migration than this for the higher versions IMHO (for correct upgrades) and then this one should still exist but would fail on existing installations. So it NEEDS to check for existence of the table.

**Pull request merged**